internal/driver/glfw/menu_darwin.go

```diff
@@ -138,13 +138,18 @@ func exceptionCallback(e *C.char) {
 }
 
 func handleSpecialItems(w *window, menu *fyne.Menu, nextItemID int, addSeparator bool) (*fyne.Menu, int) {
-	for i, item := range menu.Items {
+	menu = fyne.NewMenu(menu.Label, menu.Items...) // copy so we can manipulate
+	for i := 0; i < len(menu.Items); i++ {
+		item := menu.Items[i]
 		switch item.Label {
 		case "Settings", "Settings…", "Preferences", "Preferences…":
 			items := make([]*fyne.MenuItem, 0, len(menu.Items)-1)
 			items = append(items, menu.Items[:i]...)
-			items = append(items, menu.Items[i+1:]...)
+			if i < len(menu.Items)-1 {
+				items = append(items, menu.Items[i+1:]...)
+			}
```

Comment:
This condition is not necessary. I vote to remove it again.

Reply:
Also, the two cases of the switch contain the same code, AFAICS. That should be consolidated.

Reply:
And: Where are the tests?

Please add an “About” case to the existing test in `menu_darwin_test.go`.
Also, I’d really hoped for a test checking the flaws that were introduced without being detected.

Reply:
I reverted the code to the older recursive version - some of the test cases seemed to only work that way. About is now added to tests